app/src/features/surveys/view/survey-animals/TelemetryDeviceForm.tsx

```diff
@@ -158,12 +175,10 @@ const DeviceFormSection = ({ values, index, mode, removeAction }: IDeviceFormSec
         </Typography>
         <Grid container spacing={3}>
           <Grid item xs={6}>
-            <CustomNumberField
+            <CustomTextField
```

Comment:
It seems that this field isn't always validating properly. Sometimes when you enter a new value, it doesn't if the device already exists and will allow a submission resulting in 500 error. Another thing that could be fixed is that when editing a device / deployment, this field is enabled, but maybe it should be disabled like with device id.